File: app/src/main/java/com/john/kot/mvvm/dongnao/viewmodel/MyViewModel.java
package com.john.kot.mvvm.dongnao.viewmodel;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;

import com.android.util.base.ViewModel;

//public class MyViewModel  extends ViewModel {
//    public MyViewModel(Application application){
//        super();
//    }
//
//    public int number;
//
//}

public class MyViewModel extends AndroidViewModel {
    public MyViewModel(Application application) {
        super(application);
    }

    public int number;
}

File: app/src/main/java/com/john/kot/mvvm/dongnao/viewmodel/MyViewModel.kt
package com.john.kot.mvvm.dongnao.viewmodel

import android.util.Log
import androidx.lifecycle.SavedStateHandle
import androidx.lifecycle.ViewModel

class MyViewModel(val savedState: SavedStateHandle) : ViewModel() {
    val TAG = "MyViewModel"

    companion object {
        val NAME_KEY = "name"
    }

    var number = 0

    fun getName() {
        val get = savedState.get<String>(NAME_KEY)
        Log.i(TAG, "getName: $get")
    }


}